src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -141,6 +135,20 @@ private void DualModeConnect_IPAddressToHost_Helper(IPAddress connectTo, IPAddre
                 Assert.True(socket.Connected);
             }
         }
+
+        private void DualModeConnect_IPAddressToHost_Fails_Helper(IPAddress connectTo, IPAddress listenOn)
+        {
+            Assert.ThrowsAny<SocketException>(() =>
+            {
+                DualModeConnect_IPAddressToHost_Helper(connectTo, listenOn, false);
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // On Unix, socket assignment is random (not incremental) and there is a small chance the
+                    // listening socket was created in another test currently running. Try the test one more time.
+                    DualModeConnect_IPAddressToHost_Helper(connectTo, listenOn, false);
```

Comment:
This is the failure\negative test, so we expect DualModeConnect_IPAddressToHost_Helper to throw SocketException. If it doesn't, and returns without an exception, then we try again and hope for the SocketException. If some other exception is thrown, then we fail the test.

E.g. here's test output encountering this issue:
Assert.Throws() Failure\nExpected: typeof(System.Net.Sockets.SocketException)\nActual: (No exception was thrown)